src/ripple/app/tx/impl/SusPay.cpp

```diff
@@ -103,8 +101,10 @@ namespace ripple {
         If the SusPay ledger entry specifies sfCancelAfter, the
         transaction will fail if sfCancelAfter <= parentCloseTime.
 
-        NOTE: It must always be possible to verify the condition
-              without retrieving the SusPay ledger entry.
+        NOTE: The reason the condition must be specified again
+              is because it must always be possible to verify
+              the condition without retrieving the SusPay
+              ledger entry.
```

Comment:
Replace the call on line 341 with `validateTrigger(*fulfillment, filfillment->condition())` and the crytopgraphic validation still happens in `preflight`.

I had not thought about the cost of serialization. You're right, we'd either deserialize the condition in the ledger or serialize the fulfillment condition. Serializing seems relatively inexpensive. Still too costly for `doApply`?